### ISSUE TITLE
Exception using delete() method

### ISSUE THREAD

**Issue by Benco1** [ecederstrand/exchangelib#262]
This module has been working just fine and then all of a sudden I'm getting errors while trying to clean up (delete) individual e-mails.

```

Traceback (most recent call last):
--
  | File "do_job.py", line 59, in <module>
  | item.delete()
  | File "/usr/local/lib/python2.7/site-packages/exchangelib/items.py", line 293, in delete
  | affected_task_occurrences=affected_task_occurrences, suppress_read_receipts=suppress_read_receipts)
  | File "/usr/local/lib/python2.7/site-packages/exchangelib/items.py", line 306, in _delete
  | raise res[0]
  | exchangelib.errors.ErrorCannotDeleteObject: Object cannot be deleted.

```

Any idea? Did they change something in the API?

**Comment by ecederstrand** [comment]
I've only seen `ErrorCannotDeleteObject` when there was an actual problem deleting the item. Maybe this specific item is locked somehow, or maybe you are denied due to some policy on the server. You could check `effective_rights` on the item. Maybe it contains a clue.

**Comment by Benco1** [comment]
Thanks for that tip. This is very weird. It's been working fine for a few weeks (i.e., able to fetch the latest e-mail, process the attachment and then do `item.delete()` with no problem). All of a sudden I'm getting that error. It seems to be flip-flopping back and forth over the weekend. From checking what you suggest I am currently getting: `effective_rights: EffectiveRights(False, False, False, True, True, True, True)`

**Comment by Benco1** [comment]
Latest:

```
Traceback (most recent call last):
--
  | File "do_job.py", line 59, in <module>
  | item.delete()
  | File "/usr/local/lib/python2.7/site-packages/exchangelib/items.py", line 293, in delete
  | affected_task_occurrences=affected_task_occurrences, suppress_read_receipts=suppress_read_receipts)
  | File "/usr/local/lib/python2.7/site-packages/exchangelib/items.py", line 303, in _delete
  | affected_task_occurrences=affected_task_occurrences, suppress_read_receipts=suppress_read_receipts)
  | File "/usr/local/lib/python2.7/site-packages/exchangelib/account.py", line 334, in bulk_delete
  | suppress_read_receipts=suppress_read_receipts,
  | File "/usr/local/lib/python2.7/site-packages/exchangelib/services.py", line 445, in _pool_requests
  | for elem in elems:
  | File "/usr/local/lib/python2.7/site-packages/exchangelib/services.py", line 272, in _get_elements_in_response
  | container_or_exc = self._get_element_container(message=msg, name=self.element_container_name)
  | File "/usr/local/lib/python2.7/site-packages/exchangelib/services.py", line 245, in _get_element_container
  | self._raise_errors(code=response_code, text=msg_text, msg_xml=msg_xml)
  | File "/usr/local/lib/python2.7/site-packages/exchangelib/services.py", line 262, in _raise_errors
  | raise vars(errors)[code](text)
  | exchangelib.errors.ErrorMailboxStoreUnavailable: The mailbox database is temporarily unavailable., The process failed to get the correct properties.
```
Is this maybe on their side?

**Comment by ecederstrand** [comment]
Yes, this looks very much like intermittent problems on the server. I've seen this lots of times on lage installations where new servers are provisioned, accounts are transferred, etc. You should contact your provider if it persists.

**Comment by ecederstrand** [comment]
Closing since I believe this is a server issue. Feel free to reopen if you think otherwise :-)

**Comment by Benco1** [comment]
@ecederstrand just wanted to follow up on this. We found out that  ...

> In Exchange Online, the default limits for the Recoverable Items quota are the same as Exchange 2016; a soft limit of 20 GB and a hard limit of 30 GB

We're using the Inbox as a queue to process e-mail attachments. Fetch 1 e-mail --> process --> delete. This use case caused us to hit those recoverable item quota limits easily over a few days and then get stuck there until the recoverable item hold was released (14 days by default).

Our admin decreased the retention time from 14 days to 1, so this seems to have solved it for us.
